docs/source/user_guides/basic_conceptions.rst

```diff
@@ -139,165 +148,185 @@ That's what the agent consists of:
-  visited node. It utilizes ``labels`` attribute of the ``Context`` object.
+    check_happy_path(pipeline, happy_path)
 
-.. note::
+Monitoring and Analytics
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-    See `documentation of Context object`_.
+Setting up bot performance monitoring and usage analytics is essential to monitor its operation and identify potential issues. 
+Monitoring helps you understand how users are interacting with the bot and whether any improvements are needed.
+Analytics data can provide valuable insights for refining the bot's behavior and responses.
 
-* Transitions were changed: transitions to next, previous and current node were replaced with special
-  standard transitions.
+DFF provides a `statistics` module as an out-of-the-box solution for collecting arbitrary statistical metrics
+from your service. Setting up the data collection is as easy as instantiating the relevant class in the same
+context with the pipeline. 
+What's more, the data you obtain can be visualized right away using Apache Superset as a charting engine.
 
 .. note::
 
-    See `tutorial on transitions`_.
+    More information is available in the respective tutorial.
+    TODO: insert link
 
-* Conditions were changed: now regular expressions are used to check user text input value.
+Iterative Improvement
+~~~~~~~~~~~~~~~~~~~~~
 
-.. note::
+To continually enhance your chat-bot's performance, monitor user feedback and analyze data on bot usage.
+For instance, the statistics or the charts may reveal that some flow is visited by users more frequently or
+less frequently than planned. This would mean that adjustments to the transition structure
+of the graph need to be made.
 
-    See `tutorial on conditions`_.
+Gradually improve the transition logic and response content based on the data received. 
+This iterative approach ensures that the bot becomes more effective over time.
 
-Further exploration
-~~~~~~~~~~~~~~~~~~~
+Data Protection
+~~~~~~~~~~~~~~~
 
-There are still a lot of capabilities of Dialog Flow Framework that remain uncovered by this tutorial.
+Data protection is a critical consideration in bot development, especially when handling sensitive information.
 
-For example:
+.. note::
 
-* You can use ``GLOBAL`` transitions that will be available from every node in your script.
-  See `tutorial on global transitions`_.
+    The DFF framework helps ensure the safety of your application by storing the history and other user data present
+    in the `Context` object under unique ids and abstracting the storage logic away from the user interface.
+    As a result, it offers the basic level of data protection making it impossible to gain unlawful access to personal information.
 
-* You can serialize context (available on every transition and response)
-  to json or dictionary in order to debug it or extract some values.
-  See `tutorial on context serialization`_.
+Documentation
```

Comment:
Do we need this section?

Reply:
It was there in the initial plan. We can expand it so that it actually starts making sense